auth/genericoauth/authority_verifier.go

```diff
@@ -0,0 +1,74 @@
+package genericoauth
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+
+	"golang.org/x/oauth2"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc/auth/verifier"
+)
+
+type AuthorityVerifier struct {
+	authority string
+}
+
+func NewAuthorityVerifier(
+	authority string,
+) verifier.Verifier {
+	return AuthorityVerifier{
+		authority: authority,
+	}
+}
+
+type GenericOAuthToken struct {
```

Comment:
Don't think this'll work as the same type name is already used here within the same package: https://github.com/concourse/atc/blob/ddbaaf7f2f30606dbc3db950c07f24a30d8f7d32/auth/genericoauth/scope_verifier.go#L28

Also, there's a lot of duplication between this verifier and the scope verifier - could this be a refactoring opportunity?